confmap/provider/envprovider/provider.go

```diff
@@ -9,28 +9,41 @@ import (
 	"os"
 	"strings"
 
+	"go.uber.org/zap"
+
 	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/confmap/provider/internal"
 )
 
 const schemeName = "env"
 
-type provider struct{}
+type provider struct {
+	logger *zap.Logger
+}
 
 // NewWithSettings returns a new confmap.Provider that reads the configuration from the given environment variable.
 //
 // This Provider supports "env" scheme, and can be called with a selector:
 // `env:NAME_OF_ENVIRONMENT_VARIABLE`
-func NewWithSettings(confmap.ProviderSettings) confmap.Provider {
-	return &provider{}
+func NewWithSettings(a confmap.ProviderSettings) confmap.Provider {
+	return &provider{
+		logger: a.Logger,
+	}
 }
 
 func (emp *provider) Retrieve(_ context.Context, uri string, _ confmap.WatcherFunc) (*confmap.Retrieved, error) {
 	if !strings.HasPrefix(uri, schemeName+":") {
 		return nil, fmt.Errorf("%q uri is not supported by %q provider", uri, schemeName)
 	}
+	envVarName := uri[len(schemeName)+1:]
+	val, exists := os.LookupEnv(envVarName)
+	if !exists {
+		emp.logger.Warn(fmt.Sprintf("Environment variable %s is used in configuration but is unset", envVarName))
+	} else if len(val) == 0 {
+		emp.logger.Warn(fmt.Sprintf("Environment variable %s is used in configuration but is empty", envVarName))
+	}
```

Comment:
The Collector structure logging approach is to have the same message always and keep parts that differ in named fields. 

Also, I am fine logging something when you have an empty variable but I would not make it a warning, maybe an info (or even debug?).

```suggestion
	if !exists {
		emp.logger.Warn("Configuration references unset environment variable", zap.String("name", envVarName))
	} else if len(val) == 0 {
		emp.logger.Info("Configuration references empty environment variable", zap.String("name", envVarName))
	}
```